internal/registration/registration.go

```diff
@@ -82,30 +121,77 @@ func (r *Registration) registerDeviceWithRetries(interval int64) {
 }
 
 func (r *Registration) registerDeviceOnce() error {
+
+	key, csr, err := r.renewCertificate()
+	if err != nil {
+		return err
+	}
+
 	hardwareInformation, err := r.hardware.GetHardwareInformation()
 	if err != nil {
 		return err
 	}
+
 	registrationInfo := models.RegistrationInfo{
-		Hardware: hardwareInformation,
+		Hardware:           hardwareInformation,
+		CertificateRequest: string(csr),
 	}
+
 	content, err := json.Marshal(registrationInfo)
 	if err != nil {
 		return err
 	}
+
 	data := &pb.Data{
 		MessageId: uuid.New().String(),
 		Content:   content,
 		Directive: "registration",
 	}
-
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 
 	// Call "Send"
-	if _, err := r.dispatcherClient.Send(ctx, data); err != nil {
+	response, err := r.dispatcherClient.Send(ctx, data)
```

Comment:
How are we going to handle registration approval? There could a lot of time before admin approves registration.

Reply:
The device will be "registerted" but now allowed to pull info/config, etc.. ;-)

Reply:
Discussed offline, let's move this in another task https://issues.redhat.com/browse/ECOPROJECT-219